python/lsst/cp/pipe/cpFunctions.py

```diff
@@ -0,0 +1,571 @@
+
+    Parameters
+    ----------
+    kwargs : `dict`
+        Parameters to initialize the blessing process.  Keys:
+
+        ``"butler"``
+            Data butler to use.
+        ``"inputCollection"``
+            Data collection to pull calibrations from.
+        ``"outputCollection"``
+            Data collection to store final calibrations.
+    """
+    def __init__(self, **kwargs):
+        self.butler = kwargs['butler']
+        self.registry = self.butler.registry
+
+        self.inputCollection = kwargs['inputCollection']
+        self.outputCollection = kwargs['outputCollection']
+
+        self.calibrationLabel = ''
+        self.instrument = ''
+
+    def findInputs(self, datasetTypeName, inputDatasetTypeName=None):
+        """Find and prepare inputs for blessing.
+
+        Parameters
+        ----------
+        datasetTypeName : `str`
+            Dataset that will be blessed.
+        inputDatasetTypeName : `str`, optional
+            Dataset name for the input datasets.  Default to
+            datasetTypeName + "Proposal".
+
+        Raises
+        ------
+        RuntimeError
+            Raised if no input datasets found or if the calibration
+            label exists and is not empty.
+        """
+        if inputDatasetTypeName is None:
+            inputDatasetTypeName = datasetTypeName + "Proposal"
+
+        self.inputValues = list(self.registry.queryDatasets(inputDatasetTypeName,
+                                                            collections=[self.inputCollection]))
+        if len(self.inputValues) == 0:
+            raise RuntimeError(f"No inputs found for dataset {inputDatasetTypeName} "
+                               f"in {self.inputCollection}")
+
+        # Construct calibration label and choose instrument to use.
+        self.calibrationLabel = f"{datasetTypeName}/{self.inputCollection}"
+        self.instrument = self.inputValues[0].dataId['instrument']
+
+        # Prepare combination of new data ids and object data:
+        self.newDataId = [value.dataId for value in self.inputValues]
+        self.objects = [self.butler.get(value) for value in self.inputValues]
+
+    def registerCalibrations(self, datasetTypeName):
+        """Add blessed inputs to the output collection.
+
+        Parameters
+        ----------
+        datasetTypeName : `str`
+            Dataset type these calibrations will be registered for.
+        """
+        # Find/make the run we will use for the output
+        run = self.registry.getRun(collection=self.outputCollection)
+        if run is None:
+            run = self.registry.makeRun(self.outputCollection)
+        self.butler.run = run
+        self.butler.collection = None
+
+        for newId, data in zip(self.newDataId, self.objects):
+            # Special case known special storageClasses.
+            if datasetTypeName in ('bias', 'dark'):
+                data = data.getImage()
+            elif datasetTypeName in ('flat'):
```

Comment:
I think you need a comma before the `)` here else you are looking for substrings within "flat" and not whether "flat" is in the tuple.